File: frontend/src/objects/order/afterSales/SubmitAfterSalesRequest.ts
import type {
  AfterSalesRequestType,
  CurrencyCents,
  ReasonText,
} from '@/objects/core/SharedObjects'

export type SubmitAfterSalesRequest = {
  requestType: AfterSalesRequestType
  reason: ReasonText
  expectedCompensationCents?: CurrencyCents
}

File: backend/src/main/scala/services/order/objects/apiTypes/SubmitAfterSalesRequest.scala
package services.order.objects.apiTypes

// Business note: protocol DTO shared with the frontend; keep field names and value object types mirrored across stacks.
import services.order.objects.*

import system.objects.given

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.*
import system.objects.*

final case class SubmitAfterSalesRequest(
    requestType: AfterSalesRequestType,
    reason: ReasonText,
    expectedCompensationCents: Option[CurrencyCents],
)
object SubmitAfterSalesRequest:
  given Encoder[SubmitAfterSalesRequest] = deriveEncoder
  given Decoder[SubmitAfterSalesRequest] = deriveDecoder
